game_server/src/collide.rs

```diff
@@ -0,0 +1,74 @@
+use crate::maths::{Vec2, Box, RBox};
+
+pub trait Collide<Rhs> {
+    fn collides(&self, other: &Rhs) -> bool;
+}
+
+impl Collide<Vec2> for Vec2 {
+    fn collides(&self, other: &Self) -> bool {
+        self == other
+    }
+}
+
+impl Collide<Vec2> for Box {
+    fn collides(&self, other: &Vec2) -> bool {
+        self.pos < *other && other < &(self.pos + self.size) 
+    }
+}
+
+impl Collide<Box> for Box {
+    fn collides(&self, other: &Self) -> bool {
```

Comment:
over-complicated code could be:
1. simplified
2. optimized
see AABB collision in the web.